### ISSUE TITLE
[Bug] Allows write of column names with `.` but breaks on read

### ISSUE THREAD

**Issue by srdsam** [single-cell-data/TileDB-SOMA#2881]
**Describe the bug**
An edge case, where users can write `obs` or `var` dataframes with columns that contain a full stop (e.g. `test.field`). However, SOMA does not support reading from these columns. Error only pops up when trying to read the `Query` object.

See error when reading her:
```
 tiledbsoma._exception.SOMAError: SOMAError: Incorrect type for attribute name: Constant(value=5)
 ```

**To Reproduce**
```python
import numpy as np
import pyarrow as pa
import pandas as pd
import os
import scanpy as sc
import tiledbsoma.io as somaio
from tiledbsoma import Experiment, AxisQuery
from tiledbsoma import Experiment

OBS_FIELDS = [
    pa.field("soma_joinid", pa.int64(), nullable=False),
    ("obs_id", pa.large_string()),
    pa.field("differentiation_day", pa.int64(), nullable=True),
    pa.field("test.field", pa.int64(), nullable=True),
]

VAR_FIELDS = [
    pa.field("soma_joinid", pa.int64(), nullable=False),
    ("var_id", pa.large_string()),
]

X_LAYERS = {"raw": pa.float32()}

OBS_TABLE_SPEC = pa.schema(OBS_FIELDS)
VAR_TABLE_SPEC = pa.schema(VAR_FIELDS)

LOCAL_PATH = os.path.join(os.getcwd(),'tests/local_atlas')

def test_correct_name():
    obs_data = {
        "soma_joinid": np.arange(10),
        "obs_id": [f"cell_{i}" for i in range(10)],
        "differentiation_day": np.random.randint(0, 10, size=10),
        "test_field": np.arange(10),
    }
    var_data = {
        "soma_joinid": np.arange(5),
        "var_id": [f"gene_{j}" for j in range(5)],
    }
    data = np.random.rand(10, 5)
    adata = sc.AnnData(X=data, obs=pd.DataFrame(obs_data), var=pd.DataFrame(var_data))
    
    uri = somaio.from_anndata(
        os.path.join(LOCAL_PATH, 'test_correct_name'),
        adata,
        'RNA',
    )
    
    with Experiment.open(uri) as exp:
        query = exp.axis_query(measurement_name='RNA', obs_query=AxisQuery(value_filter="test_field==5"))
        assert query.n_obs == 1
        
def test_column_naming_bug():
    obs_data = {
        "soma_joinid": np.arange(10),
        "obs_id": [f"cell_{i}" for i in range(10)],
        "differentiation_day": np.random.randint(0, 10, size=10),
        "test.field": np.arange(10),
        "test_field": np.arange(10),
    }
    var_data = {
        "soma_joinid": np.arange(5),
        "var_id": [f"gene_{j}" for j in range(5)],
    }
    data = np.random.rand(10, 5)
    adata = sc.AnnData(X=data, obs=pd.DataFrame(obs_data), var=pd.DataFrame(var_data))
    
    uri = somaio.from_anndata(
        os.path.join(LOCAL_PATH, 'test_column_naming_bug'),
        adata,
        'RNA',
    )
    
    with Experiment.open(uri) as exp:
        query = exp.axis_query(measurement_name='RNA', obs_query=AxisQuery(value_filter="test.field==5"))
        assert query.n_obs == 1
```
Error:
```python
_______________________________________________________________________ test_column_naming_bug ________________________________________________________________________

    def test_column_naming_bug():
        obs_data = {
            "soma_joinid": np.arange(10),
            "obs_id": [f"cell_{i}" for i in range(10)],
            "differentiation_day": np.random.randint(0, 10, size=10),
            "test.field": np.arange(10),
            "test_field": np.arange(10),
        }
        var_data = {
            "soma_joinid": np.arange(5),
            "var_id": [f"gene_{j}" for j in range(5)],
        }
        data = np.random.rand(10, 5)
        adata = sc.AnnData(X=data, obs=pd.DataFrame(obs_data), var=pd.DataFrame(var_data))
    
        uri = somaio.from_anndata(
            os.path.join(LOCAL_PATH, 'test_column_naming_bug'),
            adata,
            'RNA',
        )
    
        with Experiment.open(uri) as exp:
            query = exp.axis_query(measurement_name='RNA', obs_query=AxisQuery(value_filter="test.field==5"))
>           assert query.n_obs == 1

tests/test_column_naming.py:76: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.9/site-packages/somacore/query/query.py:186: in n_obs
    return len(self.obs_joinids())
.venv/lib/python3.9/site-packages/somacore/query/query.py:171: in obs_joinids
    return self._joinids.obs
.venv/lib/python3.9/site-packages/somacore/query/query.py:711: in obs
    self._cached_obs = _load_joinids(
.venv/lib/python3.9/site-packages/somacore/query/query.py:735: in _load_joinids
    tbl = df.read(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <DataFrame 'file:///home/ec2-user/core/insitro_core/atlas/tests/local_atlas/test_column_naming_bug/obs' (CLOSED for 'r')>, coords = ()
column_names = ['soma_joinid']

    def read(
        self,
        coords: options.SparseDFCoords = (),
        column_names: Optional[Sequence[str]] = None,
        *,
        result_order: options.ResultOrderStr = options.ResultOrder.AUTO,
        value_filter: Optional[str] = None,
        batch_size: options.BatchSize = _UNBATCHED,
        partitions: Optional[options.ReadPartitions] = None,
        platform_config: Optional[options.PlatformConfig] = None,
    ) -> TableReadIter:
        """Reads a user-defined subset of data, addressed by the dataframe indexing columns,
        optionally filtered, and return results as one or more `Arrow tables <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_.
    
        Args:
            coords:
                For each index dimension, which rows to read.
                Defaults to ``None``, meaning no constraint -- all IDs.
            column_names:
                The named columns to read and return.
                Defaults to ``None``, meaning no constraint -- all column names.
            result_order:
                Order of read results.
                This can be one of 'row-major', 'col-major', or 'auto'.
            value_filter:
                An optional [value filter] to apply to the results.
                Defaults to no filter.
            partitions:
                An optional :class:`ReadPartitions` hint to indicate
                how results should be organized.
    
        Returns:
            A :class:`TableReadIter` that can be used to iterate through the result set.
    
        Raises:
            SOMAError:
                If ``value_filter`` can not be parsed.
            ValueError:
                If ``coords`` are malformed or do not index this DataFrame.
            SOMAError:
                If the object is not open for reading.
    
        Notes:
            The ``coords`` parameter will support, per dimension:
            a list of values of the type of the indexed column.
    
            Acceptable ways to index:
    
            * A sequence of coordinates is accepted, one per dimension.
            * Sequence length must be <= number of dimensions.
            * If the sequence contains missing coordinates (length less than number of dimensions),
              then ``slice(None)`` -- i.e. no constraint -- is assumed for the missing dimensions.
            * Per-dimension, explicitly specified coordinates can be one of: None, a value, a
              list/``numpy.ndarray``/``pyarrow.Array``/etc of values, a slice, etc.
            * Slices are doubly inclusive: ``slice(2,4)`` means [2,3,4] not [2,3].
              Slice steps are not supported.
              Slices can be ``slice(None)``, meaning select all in that dimension, and may be half-specified,
              e.g.  ``slice(2,None)`` or ``slice(None,4)``.
            * Negative indexing is unsupported.
    
        Lifecycle:
            Experimental.
        """
        del batch_size  # Currently unused.
        _util.check_unpartitioned(partitions)
        self._check_open_read()
    
        handle = self._handle._handle
    
        context = handle.context()
        if platform_config is not None:
            config = context.tiledb_config.copy()
            config.update(platform_config)
            context = clib.SOMAContext(config)
    
        sr = clib.SOMADataFrame.open(
            uri=handle.uri,
            mode=clib.OpenMode.read,
            context=context,
            column_names=column_names or [],
            result_order=_util.to_clib_result_order(result_order),
            timestamp=handle.timestamp and (0, handle.timestamp),
        )
    
        if value_filter is not None:
>           sr.set_condition(QueryCondition(value_filter), handle.schema)
E           tiledbsoma._exception.SOMAError: SOMAError: Incorrect type for attribute name: Constant(value=5)
```
**Versions (please complete the following information):**
```
python = ">=3.9,<4.0"
tiledbsoma = "==1.11.4"
```
Tested on an ubuntu OS, also noticed error on a MacOS.

**Additional context**

**Comment by johnkerl** [comment]
Hi @srdsam ! This is addressed on #2865 which will be in our next TileDB-SOMA release -- which will be either 1.13.1 or 1.14.0.

The TL;DR is that
```
query = exp.axis_query(measurement_name='RNA', obs_query=AxisQuery(value_filter="test.field==5"))
```
needs to be
```
query = exp.axis_query(measurement_name='RNA', obs_query=AxisQuery(value_filter='attr("test.field")==5'))
```

This is because the query-parser is using the Python syntax parser, and `test.field` -- while a valid identifier in R -- is not a valid identifer in Python.

The PR #2865 offers more transparent guidance here.

**Comment by srdsam** [comment]
Thanks @johnkerl !
